contracts/Gate.sol

```diff
@@ -0,0 +1,136 @@
+        require(bosonRouterAddress != address(0), "UNSPECIFIED_BR"); //hex"20" FISSION.code(FISSION.Category.Find, FISSION.Status.NotFound_Unequal_OutOfRange)
+        require(msg.sender == bosonRouterAddress, "UNAUTHORIZED_BR"); //hex"10" FISSION.code(FISSION.Category.Permission, FISSION.Status.Disallowed_Stop)
+        _;
+    }
+
+    /**
+     * @notice For a given _tokenIdSupply, it tells on which NFT it depends
+     * @param _tokenIdSupply an ID of a supply token (ERC-1155) [voucherSetID]
+     * @return quest NFT token ID
+     */
+    function getNftTokenId(uint256 _tokenIdSupply) external view override returns (uint256) {
+        return voucherToToken[_tokenIdSupply];
+    }
+
+    /**
+     * @notice Sets the contract, where gate contract checks if quest NFT token exists
+     * @param _nonTransferableTokenContractAddress address of a non-transferable token contract
+     */
+    function setNonTransferableTokenContract(
+        address _nonTransferableTokenContractAddress
+    ) external override onlyOwner {
+        nonTransferableTokenContract = IERC1155(
+            _nonTransferableTokenContractAddress
+        );
+
+        emit LogNonTransferableContractSet(_nonTransferableTokenContractAddress);
+    }
+
+    /**
+     * @notice Sets the Boson router contract address, from which deactivate is accepted
+     * @param _bosonRouterAddress address of a non-transferable token contract
+     */
+    function setBosonRouterAddress(address _bosonRouterAddress)
+        external
+        override
+        onlyOwner
+    {
+        bosonRouterAddress = _bosonRouterAddress;
+
+        emit LogBosonRouterSet(_bosonRouterAddress);
+    }
+
+    /**
+     * @notice Registers connection between setID and tokenID
+     * @param _tokenIdSupply an ID of a supply token (ERC-1155) [voucherSetID]
+     * @param _nftTokenID an ID of a quest token
+     */
+    function registerVoucherSetID(uint256 _tokenIdSupply, uint256 _nftTokenID)
+        external
+        override
+        whenNotPaused
+        onlyOwner
+    {
+        require(_nftTokenID != 0, "TOKEN_ID_0_NOT_ALLOWED");
+        require(_tokenIdSupply != 0, "INVALID_TOKEN_SUPPLY");
+
+        voucherToToken[_tokenIdSupply] = _nftTokenID;
+
+        emit LogVoucherSetRegistered(_tokenIdSupply, _nftTokenID);
+    }
+
+    /**
+     * @notice Checks if user posesses the required quest NFT token for given voucher set
+     * @param _user user address
+     * @param _tokenIdSupply an ID of a supply token (ERC-1155) [voucherSetID]
+     * @return true if user posesses quest NFT token, and the token is not deactivated
+     */
+    function check(address _user, uint256 _tokenIdSupply)
+        external
+        view
+        override
+        returns (bool)
+    {
+        return
+            !isDeactivated[_user][_tokenIdSupply] &&
+            nonTransferableTokenContract.balanceOf(
+                _user,
+                voucherToToken[_tokenIdSupply]
+            ) >
+            0;
```

Comment:
this should go up a line? Would be nicer to read ...

Reply:
done